canonical-web-and-design/deployment-configs
# Add the jaas-dashboard to jaas.ai

jaas-dashboard is the future of jaas.ai. For now, though, we'll just make it work by routing `/static`, `/models` and `/controllers` to the `jaas-dashboard` app.

To do this, I've extended the `routes:` blocks to support multiple paths per route, and in doing so I realised there are a bunch of possible optimisations that I think should be separately tackled in https://github.com/canonical-web-and-design/deployment-configs/issues/303.

QA
--

You'll have to first make sure an image for `jaas-dashboard` is deployed to the Canonical docker hub at `prod-comms.docker-registry.canonical.com/jaas.ai-dashboard`.

Then:

``` bash
./qa-deploy production sites/jaas.ai.yaml
watch microk8s.kubectl get all,ingress
```

And add `jaas.ai` to `/etc/hosts`:

``` bash
echo "127.0.0.1 jaas.ai" | sudo tee -a /etc/hosts
```

^ once everything is `Running` and the ingress has `ADDRESS 127.0.0.1` assigned, go to https://jaas.ai/models in a private window, and check dashboard works (although you'll have to accept the untrustworthy cert)

Now clean up `/etc/hosts`:

``` bash
sudo sed -i '/ jaas.ai/d' /etc/hosts
```
